### PR TITLE
ME-32 Add termNetDaysOptions to customer configuration

### PR DESCRIPTION
**Pull request by dhiraj-toptal**

### CODE REVIEW

File: Model/Request/Transactions.php [hunk 1]
```diff
@@ -144,4 +174,18 @@ protected function getBuyerParams(Quote $quote)
...
         return $params;
     }
+
+    /**
+     * @param $customerId
+     * @return mixed|string
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getAttributeValue($customerId)
```
Comment (ynaxonBalance) [inline Model/Request/Transactions.php]: rename to something like `getTermOptions`

File: Model/Request/Transactions.php [hunk 2]
```diff
@@ -102,24 +124,32 @@ protected function getParams()
         $quote->collectTotals();
         $requiresShipping = $quote->getShippingAddress() !== null ? 1 : 0;
         $quoteTotals = $this->_cartTotalRepository->get($quote->getId());
+        $termsOptions = $this->getAttributeValue($this->customerSession->getCustomer()->getId());
+        $options = [];
+        if (!empty($termsOptions)) {
+            foreach (explode(",", $termsOptions) as $terms) {
```
Comment (ynaxonBalance) [inline Model/Request/Transactions.php]: why not returning an array from `getAttributeValue`?

**Comment by dhiraj-toptal** [comment]
Hi @ynaxonBalance 
I have updated PR with latest code and fixes. Please review.

**Comment by ynaxonBalance** [comment]
image
Very nice! can we add a message : "Choose up to 3 terms options"

@dhiraj-toptal

**Comment by ynaxonBalance** [comment]
image
Selected terms options does not show on the customers table

**Comment by dhiraj-toptal** [comment]
HI @ynaxonBalance 
I have added notice message in admin checkbox field and on Customer list terms options are also showing. Please review.

File: Setup/Patch/Data/TermOptions.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply()
+    {
+        $this->moduleDataSetup->getConnection()->startSetup();
+        /** @var CustomerSetup $customerSetup */
+        $customerSetup = $this->customerSetupFactory->create(['setup' => $this->moduleDataSetup]);
+        $customerSetup->removeAttribute(\Magento\Customer\Model\Customer::ENTITY, 'term_options');
```
Comment (ynaxonBalance) [inline Setup/Patch/Data/TermOptions.php:54]: import (use) on top of file

**Review by ynaxonBalance — APPROVED** [review]
Nice! left one commet, please take a look before merging

**Comment by dhiraj-toptal** [comment]
Hi @ynaxonBalance 
Fixed above comments in ME-43 PR.
Merging this.

**Pull request merged**